glean-core/python/glean/_process_dispatcher.py

```diff
@@ -10,52 +10,46 @@
 This is used only by the PingUploadWorker (and the test-only feature to delete
 temporary data directories which has a potential race condition with the
 PingUploadWorker).
+
+This uses the lower level `subprocess` library rather than `multiprocessing` to
+avoid the unnecessary complexity and not place burdens on Glean's users to
+architect their application startup to be `multiprocessing` compatible on
+Windows.
 """
 
+import base64
+import pickle
+import subprocess
 import sys
-from typing import Optional, TYPE_CHECKING, Union
-
+from typing import Optional, Union
 
-if TYPE_CHECKING:
-    import multiprocessing  # noqa
 
-
-def _work_wrapper(func, args):
-    """
-    A wrapper to call a function in another process and convert its boolean
-    success value into a process exitcode.
-    """
-    success = func(*args)
-
-    if success:
-        sys.exit(0)
-    else:
-        sys.exit(1)
+from .subprocess import _process_dispatcher_helper
 
 
 class _SyncWorkWrapper:
     """
     A wrapper to synchronously call a function in the current process, but make
-    it have the same (limited) interface as if it were a
-    multiprocessing.Process:
+    it have the same (limited) interface as if it were a `subprocess.Popen`
+    object:
 
         >>> p = ProcessDispatcher.dispatch(myfunc, ())
```

Comment:
Why do the docs of `_SyncWorkWrapper` talk about `ProcessDispatcher`?

Reply:
ooooooh, this whole thing only exists so we can swap out the async vs sync worker.
That wasn't super-clear to me. Maybe worth calling out a bit in the docs?